syft/pkg/cataloger/apkdb/parse_apk_db.go

```diff
@@ -161,3 +181,42 @@ func parseApkDBEntry(reader io.Reader) (*pkg.ApkMetadata, error) {
 
 	return &entry, nil
 }
+
+func discoverPackageDependencies(pkgs []*pkg.Package) (relationships []artifact.Relationship) {
+	// map["provides" string] -> packages that provide the "p" key
+	lookup := make(map[string][]*pkg.Package)
+	// read "Provides" (p) and add as keys for lookup keys
+	for _, p := range pkgs {
+		apkg, ok := p.Metadata.(pkg.ApkMetadata)
+		if !ok {
+			// TODO warn
+			continue
+		}
+		for _, provides := range apkg.Provides {
+			// TODO: parse provides to remove =*
+			lookup[provides] = append(lookup[provides], p)
+		}
+	}
+
+	// read "Pull Dependencies" (D) and match with keys
+	for _, p := range pkgs {
+		apkg, ok := p.Metadata.(pkg.ApkMetadata)
+		if !ok {
+			// TODO warn
+			continue
+		}
+
+		for _, dep := range apkg.PullDependencies {
+			// use the lookup to find what pkg we depend on
+			for _, depPkg := range lookup[dep] {
+				// this is a pkg that package "p" depends on... make a relationship
+				relationships = append(relationships, artifact.Relationship{
+					From: p,
+					To:   depPkg,
+					Type: "", // TODO... I think "RUNTIME_DEPENDENCY_OF" ... but really HAS_RUNTIME_DEPENDENCY
```

Comment:
Another TODO here might be important

Reply:
cc @wagoodman

Reply:
we didn't land on a relationship type yet here, though it's most likely https://github.com/anchore/syft/blob/ed5c9c9b250615608da27a553e2ba404311b3d7c/syft/artifact/relationship.go#L13-L14 . The comment is referring to the direction of the edge... the relationship should read like a sentence:
```
<FROM> <TYPE> <TO>
```
so, for example:
```
[python package "requests"] is a [RUNTIME DEPENDENCY OF] [python package "anchore engine"]
```
or
```
[python package "anchore engine"] [HAS RUNTIME DEPENDENCY] [python package "requests"]
```
We should pick the direction that makes the most sense and try to stay consistent across all ecosystems.